tests/test_date.py

```diff
@@ -157,3 +157,19 @@ def test_earliest_date():
     result = earliest_date(['1686-06', '1686-06-30'])
 
     assert expected == result
+
+
+def test_earliest_date_missing_month_and_day_with_both_defaults():
+    assert earliest_date(["2000"], default_month=2, default_day=3) == "2000-02-03"
+
+
+def test_earliest_date_missing_month_and_day_with_default_month():
+    assert earliest_date(["2000"], default_month=4) == "2000-04"
+
+
+def test_earliest_date_missing_month_and_day_with_default_day():
+    assert earliest_date(["2000"], default_day=3) == "2000"
```

Comment:
this is suprising. What about adding a `full_date` boolean argument to `dumps` that uses `1` as a dummy value (I don't see why you'd ever want to use anything else) and using that in `earliest_date` instead?

Reply:
There might be the case where you want only the month, why not

Reply:
[YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)

Reply:
With only an argument `full_date` it's not possible to set default value for missing day/month

Reply:
As discussed IRL, we'll keep it simple, since there's too much complexity involved related to usual date manipulation (how many days in a month, ecc).

Reply:
As you wish.